app/models/metrics/base_metric.rb

```diff
@@ -124,5 +135,25 @@ def value_key(column, row)
     def value(column, row)
       @values[value_key(column, row)]
     end
+
+    def build_label(label, supplier)
+      if supplier.present?
+        "#{label} (#{supplier.key})"
+      else
+        label
+      end
+    end
+
+    def build_file(file, supplier)
+      if supplier.present?
+        "#{file}-#{supplier.key}"
+      else
+        file
+      end
+    end
+
+    def pause
+      sleep(rand(0..0.1).round(2))
```

Comment:
Maybe add `if Rails.env.production?` suffix to this method? Seems like there would be no need to artificially slow down development environment or test suite.

Reply:
good point - so I'll add a check for `Rails.env.test?` to not slow down tests, but I think it is necessary to have a very short pause in dev/staging/uat/pre-prod/production environments to allow the database time to recover from each query.